### ISSUE TITLE
compiler_protocol_init false positive when calling NSNumber(integerLiteral)

### ISSUE THREAD

**Issue by tbaranes** [realm/SwiftLint#1158]
After bumping to SwiftLint 0.16.0, I'm getting a `compiler_protocol_init` when using `NSNumber(integerLiteral:)` (or any `NSNumber` literal initialiser).
I have to use `NSNumber` due to some Objective-C code (a method called from swift code which is taking an array of NSNumber). 

Since there's no literal to create a `NSNumber` in swift (is there one?), I think that's a false positive 😬

**Comment by ikesyo** [comment]
You can use a number literal to create an `NSNumber` instance as follows:

```swift
let number: NSNumber = 10
// or
let number = 10 as NSNumber
```

**Comment by tbaranes** [comment]
Didn't though about this. Thanks!
